stdlib/Pkg/src/Pkg.jl

```diff
@@ -1,320 +1,78 @@
-# This file is a part of Julia. License is MIT: https://julialang.org/license
-
 __precompile__(true)
-
-"""
-    Pkg
-
-The `Pkg` module provides package management for Julia.
-Use
-`Pkg.status()` for a list of installed packages,
-`Pkg.add("<pkg name>")` to add a package,
-`Pkg.update()` to update the installed packages.
-
-Please see the manual section on packages for more information.
-"""
 module Pkg
 
-export Dir, Types, Reqs, Cache, Read, Query, Resolve, Write, Entry
-export dir, init, add, available, installed, status, clone, checkout,
-       update, resolve, test, build, free, pin, PkgError, setprotocol!
-
-const DEFAULT_META = "https://github.com/JuliaLang/METADATA.jl"
-const META_BRANCH = "metadata-v2"
-
-struct PkgError <: Exception
-    msg::AbstractString
-    ex::Union{Exception, Nothing}
-end
-PkgError(msg::AbstractString) = PkgError(msg, nothing)
-function Base.showerror(io::IO, pkgerr::PkgError)
-    print(io, pkgerr.msg)
-    if pkgerr.ex !== nothing
-        pkgex = pkgerr.ex
-        if isa(pkgex, CompositeException)
-            for cex in pkgex
-                print(io, "\n=> ")
-                showerror(io, cex)
-            end
-        else
-            print(io, "\n")
-            showerror(io, pkgex)
-        end
+import Random
+import REPL
+using REPL.TerminalMenus
+
+depots() = Base.DEPOT_PATH
+logdir() = joinpath(depots()[1], "logs")
+devdir() = get(ENV, "JULIA_PKG_DEVDIR", joinpath(homedir(), ".julia", "dev"))
+
+have_warned_session = false
+function print_first_command_header()
+    global have_warned_session
+    have_warned_session && return
+    isinteractive() || return
+    if !PKG3_IS_PRECOMPILED && !haskey(ENV, "JULIA_PKG3_DISABLE_PRECOMPILE_WARNING")
```

Comment:
Should these env. variables also be renamed to `*_PKG_*`?  (Also, presumably the binaries will be compiled with this set, right?)